src/websocket-client.js

```diff
@@ -34,6 +34,7 @@ class WebSocketClient extends Emitter {
     this.token = token;
     this.events = events;
     this.options = options;
+    this.listeners = [];
```

Comment:
`WebSocketClient` extends `Emitter` that already have it's callback mechanism. Can you use it ?

Reply:
Good point, refactored to use the `eventEmitter`